kbfsgit/runner.go

```diff
@@ -703,17 +703,22 @@ func (r *runner) processGogitStatus(ctx context.Context,
 				continue
 			}
 			switch fsEvent.EventType {
-			case libfs.FSEventLock:
-				if currStage == plumbing.StatusIndexOffset {
-					// See comment above about plumbing.StatusIndexOffset. This
-					// event happens before IndexOffset is done, so we print
-					// "done." along with a newline here and start waiting for
-					// journal.
-					if !donePrinted {
-						fmt.Fprintf(r.errput, "done.\n")
-						donePrinted = true
-					}
+			case libfs.FSEventLock, libfs.FSEventUnlock:
+				if !donePrinted {
+					fmt.Fprintf(r.errput, "done.\n")
+					donePrinted = true
+				}
+				// Since we flush all blocks in Lock, subsequent calls to
+				// Lock/Unlock normally don't take much time. So we only print
+				// journal status if it's been longer than
+				// unlockPrintBytesStatusThreshold and fsEvent.Done hasn't been
+				// closed.
+				timer := time.NewTimer(unlockPrintBytesStatusThreshold)
+				select {
```

Comment:
Might be nice to have a case for `<-ctx.Done` here, just in case (the done ch should be closed if that happens, but you never know...).

Reply:
Sure; will push a separate PR. Thanks!